docs/edgeql/paths.rst

```diff
@@ -78,58 +138,96 @@ Starting from each user, the path below traverses all *incoming* links labeled
 
 .. code-block:: edgeql
 
-  select User.<author;
+    select User.<author;
+
+This query works, showing both the ``BlogPost`` and the ``Comment`` in the
+database. However, we can't impose a shape on it:
+
+.. code-block:: edgeql
+
+    select User.<author { text };
 
 As written, EdgeDB infers the *type* of this expression to be
-:eql:type:`BaseObject`, not ``BlogPost``. Why? Because in theory, there may be
-several links named ``author`` that point to ``User``.
+:eql:type:`BaseObject`. Why? Because in theory, there may be
+several links named ``author`` from different object types
+that point to ``User``. And there is no guarantee that each 
+of these links will have a property called ``text``.
 
 .. note::
   ``BaseObject`` is the root ancestor of all object types and it only contains
   a single property, ``id``.
 
-Consider the following addition to the schema:
+As such, commonly you'll want to narrow the results to a particular type.
+To do so, use the :eql:op:`type intersection <isintersect>` operator: 
+``[is Foo]``:
+
+.. code-block:: edgeql
+    
+    # BlogPost objects that link to the user via a link named author
+    select User.<author[is BlogPost];
+    
+    # Comment objects that link to the user via a link named author
+    select User.<author[is Comment];
+
+    # All objects that link to the user via a link named author
+    select User.<author;
+
+Or parsed one step at a time, the above queries can be read as follows:
+
+.. code-block:: edgeql
+
+    # User.<        Objects that link to the user
+    # author        via a link named author
+
+    # User.<        Objects that link to the user
+    # author        via a link named author
+    # [is BlogPost] that are BlogPosts
+
+    # User.<        Objects that link to the user
+    # author        via a link named author
+    # [is Comment]  that are Comments
```

Comment:
I was concerned about the edgeql code block because that gives the user a "copy" button. I don't want to imply here that there is something a user could or should copy and paste somewhere. Almost just swapped it to a bare code block which doesn't get the copy button. Then I realized we're really just making a table here. Even though we could lay it out with a code block, it becomes less accessible than if we just make a proper table. Does that approach make sense?

```suggestion
================================ ===================================
Syntax                           Meaning
================================ ===================================
``User.<``                       Objects that link to the user
``author``                       via a link named author
================================ ===================================

================================ ===================================
Syntax                           Meaning
================================ ===================================
``User.<``                       Objects that link to the user
``author``                       via a link named author
``[is BlogPost]``                that are ``BlogPost`` objects
================================ ===================================

================================ ===================================
Syntax                           Meaning
================================ ===================================
``User.<``                       Objects that link to the user
``author``                       via a link named author
``[is Comment]``                 that are ``Comment`` objects
================================ ===================================
```

Reply:
Yes! Looks good that way.